### DISCUSSION TITLE
Could not able to config Mongodb for integration test bcs of ServerAddressMapping.serverAddress not found for flapdoodle latest version 4.6.1

### DISCUSSION THREAD

**Discussion by pspatel602** [flapdoodle-oss/de.flapdoodle.embed.mongo#454] [Q&A]
I'm trying to do the embedded config to test the transactional endpoint in spring app. I'm tying to start the embedded mongodb by using the code snippet given in the How To section's **Start mongos with mongod instance** section, but there is no mention of which imports have used, so I've look into the example given in test/example directory which is **StartConfigAndMongoDBServerTest**. In this example serverAddress imported as static method from de.flapdoodle.embed.mongo.ServerAddressMapping.serverAddress; However, I could not able to import it in flapdoodle 4.6.1 plugin in spring app. Could anyone help me with this issue please?

Java 11
Spring 2.5.14
-  * spring-data-mongodb 3.1.15
-  * spring-boot-starter-data-mongodb
-  * mongodb-springdata-v3-driver
-  * Flapdoodle version 4.6.1

**Comment by michaelmosmann** [comment]
@pspatel602 flapdoodle 4.x.x has a lot of changes so that it does not work with the old spring integration .. for spring 2.6.x, 2.7.x and 3.0.x i wrote a replacement: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo.spring .. but not for spring 2.5.x .. 

i guess that you are not about to switch to spring 2.6.x?

**Comment by michaelmosmann** [comment]
@pspatel602 i created an issue for that: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/455

**Comment by michaelmosmann** [comment]
@pspatel602 i close this discussion .. see #455

**Comment by Oskarovsky** [comment]
Hi, 

I've got the same problem. I've been trying to use spring 2.5 and 2.7 but for both was the same:
Cannot find **ServerAddressMapping** class.

Imported dependencies from de.flapdoodle:
            dependency 'de.flapdoodle.embed:de.flapdoodle.embed.mongo:4.6.2'
            dependency 'de.flapdoodle.embed:de.flapdoodle.embed.mongo.spring27x:4.6.2'

**Reply by michaelmosmann** [reply]
@Oskarovsky spring 2.7.x comes with some outdate version of flapdoodle embedded mongodb.. maybe there is some kind of class path collision.. maybe there hase something to be done: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo.spring/tree/spring-2.7.x#spring-gradle-plugin-iospringdependency-management
